src/storage/v3/mvcc.hpp

```diff
@@ -77,10 +81,18 @@ inline void ApplyDeltasForRead(Transaction *transaction, const Delta *delta, Vie
 /// transaction) and returns a `bool` value indicating whether the caller can
 /// proceed with a write operation.
 template <typename TObj>
+requires utils::SameAsAnyOf<TObj, Edge, Vertex>
 inline bool PrepareForWrite(Transaction *transaction, TObj *object) {
-  if (object->delta == nullptr) return true;
+  auto *delta_holder = std::invoke([object]() -> auto * {
+    if constexpr (std::is_same_v<TObj, Vertex>) {
+      return &object->second;
+    } else {
+      return object;
+    }
+  });
```

Comment:
Maybe make a function from this as this is used throughout this file. You might even use overloading:
```
VertexData* DeltaHolder(Vertex *vertex) {
    return vertex->second;
}

Edge* DeltaHolder(Edge* edge) {
    return edge;
}
```

Whichever is more suitable in your mind.